int32 called_count
string message
bool move_up
bool move_down
bool move_left
bool move_right
int32 move_degree
bool is_finish
